src/components/karaoke.tsx

```diff
@@ -219,8 +220,21 @@ export function Karaoke({ code }: { code: string }) {
     return match ? match[1] : null;
   };
 
+  if (loading) {
+    return (
+      <div className="flex justify-center items-center h-screen">
+        Loading session...
+      </div>
+    );
+  }
+
+  if (error) {
+    setError(error);
+    return <div className="text-red-500 text-center">{error}</div>;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There seems to be a redundant line of code at line 232 where you're setting the error state inside an if block that checks if there's an error. This is unnecessary as the error state should have been set before reaching this point.
```diff
-    setError(error);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->